python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+        run : string or int
+            Optional run number, to be used for repos containing multiple runs
+        """
+        self.log.info("Running eotest routines direct")
+        runs = butler.queryMetadata('raw', ['run'])
+        if isinstance(run, int):
+            run = str(run)
+        if len(runs) != 1 and run is None:  # lots found and we don't know which one to choose
+            raise RuntimeError("Butler query found %s for runs. eotest datasets must have a run numbers, and"
+                               " must specify which run to use if a respoitory contains multiple runs."%runs)
+        elif run is not None and run not in runs:  # Could be specifying one of many, or one of one here
+            raise RuntimeError("Butler query found %s for runs, but the run specified (%s) "
+                               "was not among them."%(runs, run))
+        elif run is None:  # we know it's OK now
+            run = butler.queryMetadata('raw', ['run'])[0]
+        del runs  # we have run defined now, so remove this to avoid potential confusion later
+
+        if not os.path.exists(self.config.eotestOutputPath):
+            os.mkdir(self.config.eotestOutputPath)
+
+        ccds = butler.queryMetadata('raw', ['ccd'])
+        imTypes = butler.queryMetadata('raw', ['imageType'])
+        testTypes = butler.queryMetadata('raw', ['testType'])
+
+        ################################
+        ################################
+        # Run the Fe55 task
+        if self.config.doFe55:
+            fe55TaskDataId = {'run': run, 'testType': 'FE55', 'imageType': 'FE55'}
+            self.log.info("Starting Fe55 pixel task")
+            for ccd in ccds:
+                if 'FE55' not in testTypes:  # TODO: Remove not to test functionality
+                    msg = "No Fe55 tests found. Available data: %s"%testTypes
+                    if self.config.requireAllEOTests:
+                        raise RuntimeError(msg)
+                    else:
+                        self.log.warn(msg + "\nSkipping Fe55 task")
+                        break
+                fe55Filenames = [butler.get('raw_filename', dataId={'visit': visit,
+                                                                    'ccd': ccd})[0][:-3]
+                                 for visit in butler.queryMetadata('raw', ['visit'], dataId=fe55TaskDataId)]
+                self.log.trace("Fe55Task: Processing %s with %s files"%(ccd, len(fe55Filenames)))
+                maskFiles = self._getMaskFiles(self.config.eotestOutputPath, ccd)
+                gains = self.fe55.run(sensor_id=ccd, infiles=fe55Filenames, mask_files=maskFiles)
+                gainsPropSet = dafBase.PropertySet()
+                for amp, gain in gains.items():  # there is no propSet.fromDict() method so make like this
+                    gainsPropSet.addDouble(str(amp), gain)
+                butler.put(gainsPropSet, 'eotest_gain', dataId={'ccd': ccd, 'run': run})
```

Comment:
Why save them as a `PropertySet`? The only reason to use that here is if you're going to need to access it from C++. Otherwise, save them as a `dict` using `PickleStorage` in the mapper policy.

Reply:
I had no idea about {{PickleStorage}} and have switched to it, it's a much better method for this, trying to force it into a {{PropertySet}} was never something I _wanted_ to do! (Seems that virtually nobody knows about this, there are only 3 uses anywhere in the stack!)